bin/kokkos_launch_compiler

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash -e
+#
+#   This script allows CMAKE_CXX_COMPILER to be a standard
+#   C++ compiler and Kokkos sets RULE_LAUNCH_COMPILE and
+#   RULE_LAUNCH_LINK in CMake so that all compiler and link
+#   commands are prefixed with this script followed by the
+#   C++ compiler. Thus if $1 == $2 then we know the command
+#   was intended for the C++ compiler and we discard both
+#   $1 and $2 and redirect the command to NVCC_WRAPPER.
+#   If $1 != $2 then we know that the command was not intended
+#   for the C++ compiler and we just discard $1 and launch
+#   the original command. Examples of when $2 will not equal
+#   $1 are 'ar', 'cmake', etc. during the linking phase
+#
+
+# if C++ is not passed, someone is probably trying to invoke it directly
+if [ -z "${1}" ]; then
+    echo -e "\n${BASH_SOURCE[0]} was invoked without the C++ compiler as the first argument."
+    echo "This script is not indended to be directly invoked by any mechanism other"
+    echo -e "than through a RULE_LAUNCH_COMPILE or RULE_LAUNCH_LINK property set in CMake\n"
+    exit 1
+fi
+
+# if there aren't two args, this isn't necessarily invalid, just a bit strange
+if [ -z "${2}" ]; then exit 0; fi
+
+# store the expect C++ compiler
+CXX_COMPILER=${1}
+
+# remove the expected C++ compiler from the arguments
+shift
+
+# after the above shift, $1 is now the exe for the compile or link command, e.g.
+#       kokkos_launch_compiler g++ gcc -c file.c -o file.o
+# becomes:
+#       kokkos_launch_compiler gcc -c file.c -o file.o
+# Check to see if the executable is the C++ compiler and if it is not, then
+# just execute the command.
+#
+# Summary:
+#       kokkos_launch_compiler g++ gcc -c file.c -o file.o
+# results in this command being executed:
+#       gcc -c file.c -o file.o
+# and
+#       kokkos_launch_compiler g++ g++ -c file.cpp -o file.o
+# results in this command being executed:
+#       nvcc_wrapper -c file.cpp -o file.o
+if [ "${CXX_COMPILER}" != "${1}" ]; then
+    # the executable is not C++ compiler so just execute the command w/o re-directing to nvcc_wrapper
+    eval $@
+else
+    # the executable is the C++ compiler, so we need to re-direct to nvcc_wrapper
+
+    # find the nvcc_wrapper from the same build/install
+    LOCAL_NVCC_WRAPPER="$(dirname ${BASH_SOURCE[0]})/nvcc_wrapper"
+
+    # find the nvcc_wrapper in the path
+    : ${BIN_NVCC_WRAPPER:-"$(which nvcc_wrapper)"}
```

Comment:
That's strange... `which nvcc_wrapper` should just return nothing if it is not in the path and and that funky bash syntax just means don't set `BIN_NVCC_WRAPPER` if it is already set or if the string is empty.

Reply:
I can't reproduce this issue. Did you do this: 
```console
: ${BIN_NVCC_WRAPPER:-"$(which nvcc_wrapper)"} 2> /dev/null
``` 
or this: 
```console
: ${BIN_NVCC_WRAPPER:-"$(which nvcc_wrapper 2> /dev/null)"}
```

Reply:
Yeah, I get an ugly block with my entire PATH and a 'no nvcc_wrapper found` error on some of our systems. Just arguing aesthetics, not function, at this point. But it would be confusing for users I think.

Reply:
The latter.

Reply:
Went with the first

Reply:
Just pulled and looks like you did the latter? I could be losing my mind.

`: ${BIN_NVCC_WRAPPER:-"$(which nvcc_wrapper 2> /dev/null)"}`

Anyhow, output is suppressed now. Looks good.

Reply:
I did the former, committed and then saw your comment and re-committed